partials/AmqpConfig.java

```diff
@@ -33,16 +27,17 @@ public class Config {
     @Value("${amqp.broker.password}")
     private String password;
 
-    {% for channelName, channel in asyncapi.channels() %}{% if channel.hasSubscribe() %}
-    @Value("${amqp.exchange.{{- channelName -}}}")
+
+    {% for channelName, channel in asyncapi.channels() %}
+    @Value("${amqp.{{- channelName -}}.exchange}")
     private String {{channelName}}Exchange;
 
-    {% endif %}{% endfor %}
-    {% for channelName, channel in asyncapi.channels() %}{% if channel.hasPublish() %}
-    @Value("${amqp.queue.{{- channelName -}}}")
+    @Value("${amqp.{{- channelName -}}.queue}")
     private String {{channelName}}Queue;
 
-    {% endif %}{% endfor %}
+    @Value("${amqp.{{- channelName -}}.routingKey}")
+    private String {{channelName}}RoutingKey;
+    {% endfor %}
```

Comment:
I'm not sure that this is a good idea to generate exchange and queue for all channels without taking into account if API really has publish\subscribe for this channel